pkg/trace/stats/aggregation.go

```diff
@@ -55,9 +53,9 @@ func getStatusCode(s *pb.Span) uint32 {
 }
 
 // NewAggregationFromSpan creates a new aggregation from the provided span and env
-func NewAggregationFromSpan(s *pb.Span, env string, agentHostname, containerID string) Aggregation {
-	synthetics := strings.HasPrefix(traceutil.GetMetaDefault(s, tagOrigin, ""), tagSynthetics)
-	hostname := traceutil.GetMetaDefault(s, tagHostname, "")
+func NewAggregationFromSpan(s *pb.Span, origin, env, agentHostname, tracerHostname, containerID string) Aggregation {
+	synthetics := strings.HasPrefix(origin, tagSynthetics)
+	hostname := tracerHostname
 	if hostname == "" {
 		hostname = agentHostname
 	}
```

Comment:
I mean, I think this whole logic can be worked out at the upmost possible level, and skip sending two args here. Instead of sending both `agentHostname` and `tracerHostname`, you could send only the final `hostname` to use.

Reply:
☝🏻 please don't forget about this. Adding it to this review too, so it doesn't get lost.